packages/@expo/fingerprint/src/Fingerprint.ts

```diff
@@ -46,15 +51,56 @@ export async function diffFingerprintChangesAsync(
 }
 
 /**
- * Differentiate two fingerprints
+ * Differentiate two fingerprints with operation type.
+ * The implementation is assumed that the sources are sorted.
```

Comment:
Is it ok for the implementation to rely on sources being sorted?

So e.g. given these sources:

```js
const file1 = {
  type: 'file',
  filePath: './assets/images/adaptive-icon.png',
  reasons: ['expoConfigExternalFile'],
  hash: '19b53640a95efdc2ccc7fc20f3ea4d0d381bb5c4',
};

const file2 = {
  type: 'file',
  filePath:
    'node_modules/expo/node_modules/@expo/config/build/resolvePackageJson.js',
  reasons: ['expoConfigPlugins'],
  hash: '0cfbcb7579dc03dde2702efe12bf88915db8b498',
};

const contents1 = {
  type: 'contents',
  id: 'packageJson:scripts',
  contents:
    '{"start":"expo start","reset-project":"node ./scripts/reset-project.js","android":"expo start --android","ios":"expo start --ios","web":"expo start --web","test":"jest --watchAll","lint":"expo lint","eas-build-post-install":"npx @expo/fingerprint . fingerprint.json"}',
  reasons: ['packageJson:scripts'],
  hash: '99d98c6c1ec629713d251a0420e9f598796fd894',
};

const contentsChanged = {
  type: 'contents',
  id: 'packageJson:scripts',
  contents:
    '{"start":"expo start","reset-project":"node ./scripts/reset-project.js","android":"expo start --android","ios":"npx expo start","web":"expo start --web","test":"jest --watchAll","lint":"expo lint","eas-build-post-install":"npx @expo/fingerprint . fingerprint.json"}',
  reasons: ['packageJson:scripts'],
  hash: 'totally-different',
};
```

We get different results for
```js
console.log(
  diffFingerprintsWithOp(
    { sources: [file1, contents1] },
    { sources: [file1, file2, contentsChanged] }
  )
);
```
vs
```js
console.log(
  diffFingerprintsWithOp(
    { sources: [file1, contents1] },
    { sources: [contentsChanged, file1, file2] }
  )
);
```

Reply:
Note - I imagine it probable _is_ okay, I just want to make sure I understand why.

Reply:
Oh I think I got it, it's because the you can't know it's a _diff_ so we can't tell what to compare it to apart from the order.

Arguably files that were added / removed could still be detected correctly though?
E.g. right not this would say that file1 was added and removed because a source was added above it 🤔 

```js
console.log(
  diffFingerprintsWithOp(
    { sources: [file1] },
    { sources: [file2, file1] }
  )
);
```

Reply:
all the sources should be sorted in a deterministic order. otherwise the final hash would be inconsistent.
e.g. [file1, file2] has different hash than [file2, file1]

the current diff implementation is based on this assumption.

Reply:
Got it, makes sense to me.